src/siminterface/simulator.py

```diff
@@ -68,13 +68,12 @@ def init(self, network_file, service_functions_file, config_file, seed):
         metrics.running_time(self.start_time, self.end_time)
         simulator_state = SimulatorState(self.network_dict, self.simulator.params.sf_placement, self.sfc_list,
                                          self.sf_list, self.traffic, self.network_stats)
+        # self.writer.write_state_results(self.env, simulator_state)
         return simulator_state
 
     def apply(self, actions: SimulatorAction):
 
-        # Add placement data to csv
-        writer.write_placement_result(self.write_stream, self.env, actions)
-
+        self.writer.write_action_result(self.env, actions)
```

Comment:
if you like you could also move passing the env to the constructor.

Reply:
That will not work because we need the env time when recording data. One way to change it is to just pass `env.now` and not the entire `env`.

Reply:
```python
def __init__(self, env):
    self.env = env

def write_action(self, actions):
    self.env.now
```